config/src/config/config_sanitizer.rs

```diff
@@ -0,0 +1,326 @@
+        chain_id: ChainId,
+    ) -> Result<(), Error> {
+        // Sanitize all of the sub-configs
+        ApiConfig::sanitize(node_config, node_role, chain_id)?;
+        BaseConfig::sanitize(node_config, node_role, chain_id)?;
+        ConsensusConfig::sanitize(node_config, node_role, chain_id)?;
+        ExecutionConfig::sanitize(node_config, node_role, chain_id)?;
+        sanitize_failpoints_config(node_config, node_role, chain_id)?;
+        sanitize_fullnode_network_configs(node_config, node_role, chain_id)?;
+        IndexerConfig::sanitize(node_config, node_role, chain_id)?;
+        IndexerGrpcConfig::sanitize(node_config, node_role, chain_id)?;
+        InspectionServiceConfig::sanitize(node_config, node_role, chain_id)?;
+        LoggerConfig::sanitize(node_config, node_role, chain_id)?;
+        MempoolConfig::sanitize(node_config, node_role, chain_id)?;
+        PeerMonitoringServiceConfig::sanitize(node_config, node_role, chain_id)?;
+        StateSyncConfig::sanitize(node_config, node_role, chain_id)?;
+        StorageConfig::sanitize(node_config, node_role, chain_id)?;
+        sanitize_validator_network_config(node_config, node_role, chain_id)?;
+
+        Ok(()) // All configs passed validation
+    }
+}
+
+/// Returns true iff failpoints are enabled
+fn are_failpoints_enabled() -> bool {
+    cfg_if! {
+        if #[cfg(feature = "failpoints")] {
+            true
+        } else {
+            false
+        }
+    }
+}
+
+/// Returns the name of the given config type
+fn get_config_name<T: ?Sized>() -> &'static str {
+    std::any::type_name::<T>()
+        .split("::")
+        .last()
+        .unwrap_or("UnknownConfig")
+}
+
+/// Validate and process the failpoints config according to the node role and chain ID
+fn sanitize_failpoints_config(
+    node_config: &mut NodeConfig,
+    _node_role: RoleType,
+    chain_id: ChainId,
+) -> Result<(), Error> {
+    let sanitizer_name = FAILPOINTS_SANITIZER_NAME.to_string();
+    let failpoints = &node_config.failpoints;
+
+    // Verify that failpoints are not enabled in mainnet
+    let failpoints_enabled = are_failpoints_enabled();
+    if chain_id.is_mainnet()? && failpoints_enabled {
+        return Err(Error::ConfigSanitizerFailed(
+            sanitizer_name,
+            "Failpoints are not supported on mainnet nodes!".into(),
+        ));
+    }
+
+    // Ensure that the failpoints config is populated appropriately
+    if let Some(failpoints) = failpoints {
+        if failpoints_enabled && failpoints.is_empty() {
+            return Err(Error::ConfigSanitizerFailed(
+                sanitizer_name,
+                "Failpoints are enabled, but the failpoints config is empty?".into(),
+            ));
+        } else if !failpoints_enabled && !failpoints.is_empty() {
+            return Err(Error::ConfigSanitizerFailed(
+                sanitizer_name,
+                "Failpoints are disabled, but the failpoints config is not empty!".into(),
```

Comment:
Is this too picky? Would someone want to turn it off (temporarily) but leave their config in place?

Reply:
Good question. I think `failpoints` are only enabled in specific types of test infra (e.g., forge), so I can't really imagine anyone wanting to enable the feature locally and not set the failpoints. I'm inclined to be strict, and then loosen it later (if we find a reasonable use-case)?